met-web/src/components/engagement/form/EngagementWidgets/WidgetsBlock.tsx

```diff
@@ -1,26 +1,31 @@
 import React, { useContext, useEffect, useState, useCallback } from 'react';
-import { Grid, Skeleton } from '@mui/material';
+import { Divider, Grid, Skeleton } from '@mui/material';
 import { MetHeader2, MetPaper, SecondaryButton } from 'components/common';
 import { WidgetCardSwitch } from './WidgetCardSwitch';
-import { If, Then, Else } from 'react-if';
+import { If, Then, Else, When } from 'react-if';
 import { WidgetDrawerContext } from './WidgetDrawerContext';
 import { ActionContext } from '../ActionContext';
 import { useAppDispatch } from 'hooks';
 import { openNotification } from 'services/notificationService/notificationSlice';
-import { Widget } from 'models/widget';
+import { Widget, WidgetType } from 'models/widget';
 import { openNotificationModal } from 'services/notificationModalService/notificationModalSlice';
 import update from 'immutability-helper';
+import { DragItem } from 'components/common/Dragndrop';
 
 const WidgetsBlock = () => {
     const { widgets, deleteWidget, updateWidgetsSorting, handleWidgetDrawerOpen, isWidgetsLoading } =
         useContext(WidgetDrawerContext);
     const { savedEngagement } = useContext(ActionContext);
     const dispatch = useAppDispatch();
 
-    const [tempWidgets, setTempWidgets] = useState<Widget[]>(widgets);
+    const [sortableWidgets, setSortableWidgets] = useState<Widget[]>([]);
+    const [fixedWidgets, setFixedWidgets] = useState<Widget[]>([]);
 
     useEffect(() => {
-        setTempWidgets(widgets);
+        const widgetsFixedList = widgets.filter((w) => w.widget_type_id == WidgetType.Phases);
```

Comment:
Maybe we can just define an array of sortable types [WidgetType.Phases, ..] and just check if type id is in it. This way we can just extend the array if mor fixed types existed

Reply:
Done